src/Core/Manifest/ClassManifest.php

```diff
@@ -644,36 +418,89 @@ public function handleFile($basename, $pathname, $depth)
             $this->classes[$lowercaseName] = $pathname;
 
             if ($extends) {
-                $extends = strtolower($extends);
+                foreach ($extends as $ancestor) {
+                    $ancestor = strtolower($ancestor);
 
-                if (!isset($this->children[$extends])) {
-                    $this->children[$extends] = array($name);
-                } else {
-                    $this->children[$extends][] = $name;
+                    if (!isset($this->children[$ancestor])) {
+                        $this->children[$ancestor] = array($className);
+                    } else {
+                        $this->children[$ancestor][] = $className;
+                    }
                 }
             } else {
-                $this->roots[] = $name;
+                $this->roots[] = $className;
             }
 
             if ($implements) {
                 foreach ($implements as $interface) {
                     $interface = strtolower($interface);
 
                     if (!isset($this->implementors[$interface])) {
-                        $this->implementors[$interface] = array($name);
+                        $this->implementors[$interface] = array($className);
                     } else {
-                        $this->implementors[$interface][] = $name;
+                        $this->implementors[$interface][] = $className;
                     }
                 }
             }
         }
 
-        foreach ($interfaces as $interface) {
-            $this->interfaces[strtolower($namespaceBase . $interface['interfaceName'])] = $pathname;
+        foreach ($interfaces as $interfaceName => $interfaceInfo) {
+            $this->interfaces[strtolower($interfaceName)] = $pathname;
+        }
+        foreach ($traits as $traitName => $traitInfo) {
+            $this->traits[strtolower($traitName)] = $pathname;
         }
-        foreach ($traits as $trait) {
-            $this->traits[strtolower($namespaceBase . $trait['traitName'])] = $pathname;
+    }
+
+    /**
+     * Clean out any information from PHP files that is nested more than $cutOffDepth in curly braces
+     *
+     * @param string $contents
+     * @param int $cutOffDepth
+     * @return string
+     */
+    protected function cleanFileContents($contents, $cutOffDepth = 1)
```

Comment:
Add a test specifically for cleanFileContents. Related to that, maybe burying it as a protected method of this class isn't as good as just splitting it out into a separate class.

Maybe `ClassContentRemover` is a better name?